pkg/database/database_integration_test.go

```diff
@@ -135,9 +166,64 @@ func TestDatabaseHandler(t *testing.T) {
 		}
 	})
 
+	t.Run("Lock/Unlock", func(t *testing.T) {
+		{
+			t.Log("Lock")
+
+			body := strings.NewReader(`{
+				"instanceId":  ` + instanceID + `
+			}`)
+			var lock model.Lock
+			client.PostJSON(t, "/databases/"+databaseID+"/lock", body, &lock)
+
+			require.Equal(t, uint(1), lock.DatabaseID)
+			require.Equal(t, uint(1), lock.InstanceID)
+			require.Equal(t, uint(1), lock.UserID)
+		}
+
+		{
+			t.Log("Unlock")
+
+			client.Delete(t, "/databases/"+databaseID+"/lock")
+
+			var database model.Database
+			client.GetJSON(t, "/databases/"+databaseID, &database)
+			require.Nil(t, database.Lock)
+		}
+	})
+
+	t.Run("ExternalDownload", func(t *testing.T) {
+		{
+			t.Log("ExternalDownload")
+
+			body := strings.NewReader(`{
+				"expiration": 60
+			}`)
+			var externalDownload model.ExternalDownload
+			client.PostJSON(t, "/databases/"+databaseID+"/external", body, &externalDownload)
+
+			require.Equal(t, uint(1), externalDownload.DatabaseID)
+			diff := int64(externalDownload.Expiration) - time.Now().Unix() - 60
+			require.Zero(t, diff)
+		}
+	})
+
 	{
 		t.Log("Delete")
 
+		// Lock database
+		body := strings.NewReader(`{"instanceId":  ` + instanceID + `}`)
+		client.PostJSON(t, "/databases/"+databaseID+"/lock", body, &model.Lock{})
+		// Attempt delete but expect a bad request response
+		client.Do(t, http.MethodDelete, "/databases/"+databaseID, nil, http.StatusBadRequest, inttest.WithAuthToken("sometoken"))
+		// Unlock database
+		client.Delete(t, "/databases/"+databaseID+"/lock")
+
+		// Create external download to ensure the database can still be deleted
+		client.PostJSON(t, "/databases/"+databaseID+"/external", strings.NewReader(`{
+				"expiration": 60
+			}`), &model.ExternalDownload{})
+
```

Comment:
Should deleting a locked db or one with external downloads be split into subtests?

Reply:
Good point! I was considering it but it'll require quite a lot of boiler plate code so I just piggy backed the clean up code for now.